packages/dx-react-grid-demos/src/demo-sources/grid-featured-chart-integration/bootstrap4/demo.jsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import { Card } from 'reactstrap';
+import {
+  RowDetailState,
+  DataTypeProvider,
+} from '@devexpress/dx-react-grid';
+import {
+  Stack,
+} from '@devexpress/dx-react-chart';
+import {
+  Chart,
+  BarSeries,
+  ArgumentAxis,
+  ValueAxis,
+  Legend,
+  Grid as ChartGrid,
+} from '@devexpress/dx-react-chart-bootstrap4';
+import {
+  Grid, Table, TableBandHeader, TableHeaderRow,
+  TableRowDetail,
+} from '@devexpress/dx-react-grid-bootstrap4';
+import { citiesCount, regionsCount } from '../../../demo-data/chart-data';
+
+const nullComponent = () => null;
+const currencyFormatter = ({ value }) => `$${value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`;
+const axisLabel = ({
+  text,
+  ...restProps
+}) => <ValueAxis.Label text={currencyFormatter({ value: text })} {...restProps} />;
+
+const CurrencyTypeProvider = props => (
+  <DataTypeProvider
+    {...props}
+    formatterComponent={currencyFormatter}
+  />
+);
+
+const legendRoot = props => (
+  <Legend.Root
+    {...props}
+    className="m-auto flex-row"
+  />
+);
+
+const barSeriesForCity = regionCities => Object
+  .keys(regionCities[0])
+  .reduce((acc, item, index) => {
+    if (item !== 'year') {
+      acc.push(
+        <BarSeries
+          key={index.toString()}
+          valueField={item}
+          argumentField="year"
+          name={item}
+        />,
+      );
+    }
+    return acc;
+  }, []);
+
+const gridDetailContainer = data => ({ row }) => {
```

Comment:
Why do you pass data from Grid? We can use `citiesCount` directly.

Reply:
Because we want to display that only one main component knowns about all data source. Children get data from the main component.

Reply:
I added `citiesCount` to state